pyoozie/tags.py

```diff
@@ -3,16 +3,31 @@
 from __future__ import unicode_literals
 from abc import ABCMeta, abstractmethod
 import re
+import string  # pylint: disable=deprecated-module
 import yattag
 
-
-MAX_IDENTIFIER_LENGTH = 39
-REGEX_IDENTIFIER = r'^[a-zA-Z_][\-_a-zA-Z0-9]{0,38}$'
+MAX_NAME_LENGTH = 255
+MAX_IDENTIFIER_LENGTH = 50
+REGEX_NAME = r'[%(chars)s]{1,%(max_length)i}' % {'chars': re.escape(string.printable), 'max_length': MAX_NAME_LENGTH}
+REGEX_IDENTIFIER = r'^[a-zA-Z_][\-_a-zA-Z0-9]{0,%i}$' % (MAX_IDENTIFIER_LENGTH - 1)
+COMPILED_REGEX_NAME = re.compile(REGEX_NAME)
 COMPILED_REGEX_IDENTIFIER = re.compile(REGEX_IDENTIFIER)
 
 
-def _validate(identifier):
+def _validate_name(name):
+    assert len(name) <= MAX_NAME_LENGTH, \
+        "Name must be less than {max_length} chars long, '{name}' is {length}".format(
+            max_length=MAX_NAME_LENGTH,
+            name=name,
+            length=len(name))
+
+    assert COMPILED_REGEX_NAME.match(name), \
+        "Name must be comprised of printable ASCII characters, '{name}' is not".format(name=name)
```

Comment:
Is a regex really the best way to check this? 
Are all printable characters really desirable (e.g. `\n`)?